core/tests/protractor_utils/StoryEditorPage.js

```diff
@@ -139,6 +198,144 @@ var StoryEditorPage = function() {
+    var warningToast = element(by.css('.toast-message'));
+    waitFor.visibilityOf(
+      warningToast,
+      'warningToast takes too long to be visible.');
+    expect(warningToast.getText()).toEqual(
+      'The given exploration already exists in the story.');
+    var closeToastButton = element(by.css('.toast-close-button'));
+    waitFor.elementToBeClickable(
+      closeToastButton,
+      'closeToastButton takes too long to be clickable.');
+    closeToastButton.click();
+  };
+
+  this.getSelectSkillModal = function() {
+    waitFor.visibilityOf(
+      selectSkillModalHeader,
+      'selectSkillModalHeader takes too long to be visible.');
+    return {
+      _searchSkillByName: function(name) {
+        waitFor.visibilityOf(
+          skillNameInputField,
+          'skillNameInputField takes too long to be visible');
+        skillNameInputField.sendKeys(name);
+      },
+
+      _selectSkillBasedOnIndex: function(index) {
+        skillListItems.then(function(elements) {
+          var selectedSkill = elements[index];
+          waitFor.elementToBeClickable(
+            selectedSkill,
+            'selectedSkill takes too long to be clickable.'
+          );
+          selectedSkill.click();
+        });
+      },
+
+      selectSkill: function(name) {
+        this._searchSkillByName(name);
+        this._selectSkillBasedOnIndex(0);
+        var doneButton = element(by.css('.btn-success'));
+        waitFor.elementToBeClickable(
+          doneButton,
+          'doneButton takes too long to be clickable');
+        doneButton.click();
+      },
+    };
+  };
+
+  this.addAcquiredSkill = function(skillName) {
+    waitFor.visibilityOf(
+      addAcquiredSkillButton,
+      'addAcquiredSkillButton takes too long to be visible');
+    waitFor.elementToBeClickable(
+      addAcquiredSkillButton,
+      'addAcquiredSkillButton takes too long to be clickable');
+    addAcquiredSkillButton.click();
+    var selectSkillModal = this.getSelectSkillModal();
+    selectSkillModal.selectSkill(skillName);
+  };
+
+  this.addPrerequisiteSkill = function(skillName) {
+    waitFor.visibilityOf(
+      addPrerequisitesSkillButton,
+      'addPrerequisitesSkillButton takes too long to be visible');
+    waitFor.elementToBeClickable(
+      addPrerequisitesSkillButton,
+      'addPrerequisitesSkillButton takes too long to be clickable');
+    addPrerequisitesSkillButton.click();
+    var selectSkillModal = this.getSelectSkillModal();
+    selectSkillModal.selectSkill(skillName);
+  };
+
+  this.deleteAcquiredSkillByIndex = function(index) {
+    deleteAcquiredSkillButton.then(function(elements) {
+      var toDelete = elements[index];
+      toDelete.click();
+    });
+  };
+
+  this.deletePrerequisiteSkillByIndex = function(index) {
+    deletePrerequisiteSkillButton.then(function(elements) {
+      var toDelete = elements[index];
+      toDelete.click();
+    });
+  };
+
+  this.expectAcquiredSkillDescriptionCardNumber = function(number) {
```

Comment:
I was a little confused by the use of `Number` at the end of this function name when reading the earlier code. I wasn't sure whether it meant the number of cards or the number of a particular card. Maybe use `Count` to disambiguate.

Alternatively, `expectNumAcquiredSkillDescriptionCards`?

Reply:
Updated